crates/bevy_pbr/src/light.rs

```diff
@@ -268,31 +266,127 @@ fn calculate_cascade_bounds(
         .collect()
 }
 
-impl CascadeShadowConfig {
-    /// Returns a cascade config for `num_cascades` cascades, with the first cascade
-    /// having far bound `nearest_bound` and the last cascade having far bound `shadow_maximum_distance`.
-    /// In-between cascades will be exponentially spaced.
-    pub fn new(
-        num_cascades: usize,
-        nearest_bound: f32,
-        shadow_maximum_distance: f32,
-        overlap_proportion: f32,
-    ) -> Self {
+/// Builder for [`CascadeShadowConfig`].
+pub struct CascadeShadowConfigBuilder {
+    num_cascades: usize,
+    minimum_distance: f32,
+    maximum_distance: f32,
+    first_cascade_far_bound: f32,
+    overlap_proportion: f32,
+}
+
+impl CascadeShadowConfigBuilder {
+    /// Constructs a new builder.
+    pub fn new() -> Self {
+        if cfg!(feature = "webgl") {
+            // Currently only support one cascade in webgl.
+            Self {
+                num_cascades: 1,
+                minimum_distance: 0.1,
+                maximum_distance: 100.0,
+                first_cascade_far_bound: 5.0,
+                overlap_proportion: 0.2,
+            }
+        } else {
+            Self {
+                num_cascades: 4,
+                minimum_distance: 0.1,
+                maximum_distance: 1000.0,
+                first_cascade_far_bound: 5.0,
+                overlap_proportion: 0.2,
+            }
+        }
+    }
+
+    /// Sets the number of shadow cascades.
+    /// More cascades increases shadow quality by mitigating perspective aliasing - a phenomenom where areas
+    /// nearer the camera are covered by fewer shadow map texels than areas further from the camera, causing
+    /// blocky looking shadows.
+    ///
+    /// This does come at the cost increased rendering overhead, however this overhead is still less
+    /// than if you were to use fewer cascades and much larger shadow map textures to achieve the
+    /// same quality level.
+    ///
+    /// In case rendered geometry covers a relatively narrow and static depth relative to camera, it may
+    /// make more sense to use fewer cascades and a higher resolution shadow map texture as perspective aliasing
+    /// is not as much an issue. Be sure to adjust `minimum_distance` and `maximum_distance` appropriately.
+    pub fn num_cascades(mut self, n: usize) -> Self {
+        self.num_cascades = n;
+        self
+    }
+
+    /// Sets the minimum shadow distance.
+    /// Areas nearer to the camera than this will likely receive no shadows.
+    /// NOTE: Due to implementation details, this usually does not impact shadow quality as much as
+    /// `first_cascade_far_bound` and `maximum_distance`.
```

Comment:
Depends on the FOV I suppose. A narrower FOV would make the view frustum body diagonal relatively larger compared to the far plane face diagonal. A wider FOV would shift their relationship in the other direction. Not sure how we can communicate this in a simple and easy way.